packages/web/src/store/hooks/scan/useScanResult.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useMemo } from 'react';
+import { useAppSelector } from '../../index';
+import { makeSelectScanResultByUrl } from '../../selectors/websiteResults';
+import { useNormalizedScanUrl } from './useNormalizedScanUrl';
+import { useScanRequest } from './useScanRequest';
+
+export const useScanResult = (
+  scanUrl: string | undefined,
+  { pollWhilePending = false, requestRescan = false } = {}
+) => {
+  const { displayUrl, normalizedUrl, parsedUrl } = useNormalizedScanUrl(scanUrl);
+
+  const scanResultSelector = useMemo(makeSelectScanResultByUrl, []);
+  const scanResult = useAppSelector((state) => scanResultSelector(state, normalizedUrl));
+
+  const requestScan = useScanRequest();
+
+  // Initial request if entity wasn't loaded
+  useEffect(() => {
+    if (normalizedUrl && !scanResult?.scan && !scanResult?.isLoading && !scanResult?.error) {
+      requestScan(normalizedUrl, requestRescan);
+    }
+  }, [normalizedUrl, scanResult, requestRescan]);
+
+  // Poll while scan is pending
+  useEffect(() => {
+    let timeoutId: number | undefined;
+    if (
+      pollWhilePending &&
+      normalizedUrl &&
+      scanResult?.scan?.status === 'pending' &&
+      !scanResult?.isLoading &&
+      !scanResult?.error
+    ) {
+      timeoutId = window.setTimeout(() => {
```

Comment:
Shouldn't it be `setInterval`?

Reply:
No, it's working as intended, as it's far easier to manage a `setTimeout` in `useEffect`, rather than a `setInterval`. Be mindful that `useEffect` is being triggered on each dependent value update, thus initiating a `setTimeout` each time a  scan is still in `pending` state and `isLoading` is no longer in effect.